pandas/core/frame.py

```diff
@@ -5537,7 +5537,8 @@ def _list_of_series_to_arrays(data, columns, coerce_float=False, dtype=None):
 def _list_of_dict_to_arrays(data, columns, coerce_float=False, dtype=None):
     if columns is None:
         gen = (list(x.keys()) for x in data)
-        columns = lib.fast_unique_multiple_list_gen(gen)
+        columns = lib.fast_unique_multiple_list_gen(
+            gen, sort=(not isinstance(data[0], OrderedDict)))
```

Comment:
should be ``not any(isinstance(d, OrderedDict) for d in data)`` I think. (a bit odd if that actually happens, though)